tools/ldb_cmd.cc

```diff
@@ -3444,88 +3444,101 @@ void DBLiveFilesMetadataDumperCommand::DoCommand() {
+        // concantenate the two string sdirectly or if we need to
+        // drop a possible extra "/" at the end of fileMetadata.db_path.
+        std::string filename =
+            blobMetadata.blob_file_path + "/" + blobMetadata.blob_file_name;
+        // Drops any repeating '/' character that could happen during
+        // concatenation of db path and file name.
+        filename = NormalizePath(filename);
+        filenametoOutputStr[filename] =
+            filename + ", column family '" + cf + "'";
+      }  // End of for-loop over blob files
+    }    // End of for-loop over column metadata
+
+    for (auto const& item : filenametoOutputStr) {
+      std::cout << item.second << std::endl;
     }
   } else {
-    std::map<std::string, std::map<int, std::vector<std::string>>>
-        filesPerLevelPerCf;
-    // Collect live files metadata.
-    // Store filenames into a 2D map, that will automatically
-    // sort by column family (first key) and by level (second key).
-    for (auto& fileMetadata : metadata) {
-      std::string cf = fileMetadata.column_family_name;
-      int level = fileMetadata.level;
-      if (filesPerLevelPerCf.find(cf) == filesPerLevelPerCf.end()) {
-        filesPerLevelPerCf.emplace(cf,
-                                   std::map<int, std::vector<std::string>>());
-      }
-      if (filesPerLevelPerCf[cf].find(level) == filesPerLevelPerCf[cf].end()) {
-        filesPerLevelPerCf[cf].emplace(level, std::vector<std::string>());
-      }
-
-      // The fileMetada.name alwasy starts with "/",
-      // however fileMetada.db_path is the string provided by
-      // the user as an input. Therefore we check if we can
-      // concantenate the two string sdirectly or if we need to
-      // drop a possible extra "/" at the end of fileMetadata.db_path.
-      std::string filename = fileMetadata.db_path + "/" + fileMetadata.name;
-      // Drops any repeating '/' character that could happen during
-      // concatenation of db path and file name.
-      filename = NormalizePath(filename);
-      filesPerLevelPerCf[cf][level].push_back(filename);
-    }
-    // For each column family,
-    // iterate through the levels and print out the live SST file names.
-    for (auto it = filesPerLevelPerCf.begin(); it != filesPerLevelPerCf.end();
-         it++) {
-      // it->first: Column Family name (string)
-      // it->second: map[level]={SST files...}.
-      std::cout << "===== Column Family: " << it->first
+    // Using GetAllColumnFamilyMetaData
+    for (auto& columnMetadata : metadata) {
+      std::cout << "===== Column Family: " << columnMetadata.name
                 << " =====" << std::endl;
 
-      // For simplicity, create reference to the inner map (level={live SST
-      // files}).
-      std::map<int, std::vector<std::string>>& filesPerLevel = it->second;
-      int maxLevel = filesPerLevel.rbegin()->first;
-
-      // Even if the first few levels are empty, they are printed out.
```

Comment:
I think this property (printing the heading even for empty levels, starting from L0) is worth preserving

Reply:
GetAllColumnFamilyMetaData returns the LevelMetadata (which contains SstFileMetadata) such that it has entry for empty levels as well (empty level before highest level in that respective column). So, we don't need to explicitly create map to track empty levels and the current change will print any empty levels.